qt/preferences_dialog.cpp

```diff
@@ -97,6 +99,28 @@ namespace qt
       });
     }
 
+    QLabel * mapLanguageLabel = new QLabel("Map Language");
+    QComboBox * mapLanguageComboBox = new QComboBox();
+    {
+      StringUtf8Multilang::Languages const & supportedLanguages = StringUtf8Multilang::GetSupportedLanguages();
+      QStringList languagesList = QStringList();
+      for (auto const & language : supportedLanguages)
```

Comment:
I switched languages in the desktop version and found a bug that incorrect language is set for some languages (e.g. Polish/Arabic, and other in the list below). Looks like the offset/lang code was shifted by 1 for some reason.

I didn't test it on Android, maybe the same issue is there too.

Reply:
This here was qt specific. Fixed.